src/compute/src/render/reduce.rs

```diff
@@ -1144,29 +1158,31 @@ where
     };
 
     let mut to_aggregate = Vec::new();
-    // First, collect all non-distinct aggregations in one pass.
-    let easy_cases = collection.explode({
-        let zero_diffs = zero_diffs.clone();
-        move |(key, row)| {
-            let mut diffs = zero_diffs.clone();
-            // Try to unpack only the datums we need. Unfortunately, since we
-            // can't random access into a Row, we have to iterate through one by one.
-            // TODO: Even though we don't have random access, we could still avoid unpacking
-            // everything that we don't care about, and it might be worth it to extend the
-            // Row API to do that.
-            let mut row_iter = row.iter().enumerate();
-            for (accumulable_index, datum_index, aggr) in simple_aggrs.iter() {
-                let mut datum = row_iter.next().unwrap();
-                while datum_index != &datum.0 {
-                    datum = row_iter.next().unwrap();
+    if simple_aggrs.len() > 0 {
```

Comment:
I need to convince myself that this is correct: The operator seems to do something when `simple_aggrs` is empty, but what exactly does it do? It seems to preserve the key, and place it adjacently of a zero diff. The zero diffs is cloned, and only when `simple_aggrs` contains any element, entries in `diffs` are replaced by non-zero elements? It might be a good idea to include a comment in the code.

Reply:
I found this a bit tricky too; here is my line of reasoning:
- we'd expect that either `simple_aggrs` or `distinct_aggrs` contain elements as per the `soft_assert_or_log` at the beginning of the function. So in normal circumstances, either the `explode` in line 1163 gets executed or the one in line 1197.
- In the exceptional circumstances where the assertion at the beginning of the function fails in a production build, then neither `explode` gets executed. But in this case, `to_aggregate` is empty, and thus collection in line 1126 is an empty collection. So there is really nothing to reduce and we will see no results out of the reduction and an ERROR in the log.

Reply:
I *think* this ends up fine when `simple_aggrs` is empty, as the diff field ends up (afaict) as a zero, and has no impact on the downstream reduce (we could replace it by `Collection::empty()`). It seems like if there are no aggregates, again the input will be empty and the output empty. That seems fine, but would be great to 1. double check, and 2. document.